src/Application.php

```diff
@@ -218,12 +163,6 @@ public function finish($callback, $priority = 0)
                 2 => $this,
             ]);
 
-            if ($ret instanceof Response) {
-                $event->setResponse($ret);
-            } elseif (null !== $ret) {
-                throw new \RuntimeException('An after middleware returned an invalid response value. Must return null or an instance of Response.');
-            }
-
```

Comment:
No, this is `finish()` (don't get fooled by the exception message which is wrong, too). `after()` still has this piece of code: https://github.com/PHP-DI/Silex-Bridge/pull/17/files#diff-c9248b3167fc44af085b81db2e292837R140